pkgs/development/compilers/go/1.11.nix

```diff
@@ -0,0 +1,194 @@
+{ stdenv, fetchFromGitHub, tzdata, iana-etc, go_bootstrap, runCommand, writeScriptBin
+, perl, which, pkgconfig, patch, procps
+, pcre, cacert, llvm
+, Security, Foundation
+, makeWrapper, git, subversion, mercurial, bazaar }:
+
+let
+
+  inherit (stdenv.lib) optionals optionalString;
+
+  clangHack = writeScriptBin "clang" ''
+    #!${stdenv.shell}
+    exec ${stdenv.cc}/bin/clang "$@" 2> >(sed '/ld: warning:.*ignoring unexpected dylib file/ d' 1>&2)
+  '';
+
+  goBootstrap = runCommand "go-bootstrap" {} ''
+    mkdir $out
+    cp -rf ${go_bootstrap}/* $out/
+    chmod -R u+w $out
+    find $out -name "*.c" -delete
+    cp -rf $out/bin/* $out/share/go/bin/
+  '';
+
+in
+
+stdenv.mkDerivation rec {
+  name = "go-${version}";
+  version = "1.11";
+
+  src = fetchFromGitHub {
+    owner = "golang";
+    repo = "go";
+    rev = "go${version}";
+    sha256 = "1k18d6rkijlgzn1zw4wphzcv6a6w9hb1msgrsh1102jb18644f2q";
+  };
+
+  GOCACHE = "off";
```

Comment:
I'm not sure this is working. The derivation is 1.1GB on my machine, and on closer inspection it seems to be mostly this directory:

```
$ du -hcs /nix/store/vk289i3s3ki84681q8pallmgcmsjg708-go-1.11/share/go/pkg/obj/go-build
673M	/nix/store/vk289i3s3ki84681q8pallmgcmsjg708-go-1.11/share/go/pkg/obj/go-build
673M	total
```

and there is a README: 

```
$ cat /nix/store/vk289i3s3ki84681q8pallmgcmsjg708-go-1.11/share/go/pkg/obj/go-build/README
This directory holds cached build artifacts from the Go build system.
Run "go clean -cache" if the directory is getting too large.
See golang.org to learn more about Go.
```

Do we want that directory as part of our go derivation?